modules/images/webp-uploads/load.php

```diff
@@ -683,3 +683,84 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
 	return rest_ensure_response( $data );
 }
 add_filter( 'rest_prepare_attachment', 'webp_uploads_update_rest_attachment', 10, 3 );
+
+/**
+ *
+ * Updates webp image when original image is edited
+ *
+ * @since n.e.x.t
+ *
+ * @param bool|null       $override  Value to return instead of saving. Default null.
+ * @param string          $file      Name of the file to be saved.
+ * @param WP_Image_Editor $image     The image editor instance.
+ * @param string          $mime_type The mime type of the image.
+ * @param int             $post_id   Attachment post ID.
+ */
+function webp_uploads_update_image_onchange( $override, $file, $image, $mime_type, $post_id ) {
+	if ( $override !== null ) {
+		return $override;
+	}
+
+	$valid_mime_transforms      = webp_uploads_get_upload_image_mime_transforms();
+	$original_directory         = pathinfo( $file, PATHINFO_DIRNAME );
+	$filename                   = pathinfo( $file, PATHINFO_FILENAME );
+	$sizes                      = get_intermediate_image_sizes();
+	$_wp_additional_image_sizes = wp_get_additional_image_sizes();
+	$target                     = ! empty( $_REQUEST['target'] ) ? preg_replace( '/[^a-z0-9_-]+/i', '', $_REQUEST['target'] ) : '';
+	$nocrop                     = false;
+	$current_mime_type          = get_post_mime_type( $post_id );
+
+	foreach ( $valid_mime_transforms[ $mime_type ] as $targeted_mime ) {
+		if ( $targeted_mime === $current_mime_type ) {
+			continue;
+		}
+
+		$allowed_mimes = array_flip( wp_get_mime_types() );
+
+		if ( ! isset( $allowed_mimes[ $targeted_mime ] ) || ! is_string( $allowed_mimes[ $targeted_mime ] ) ) {
+			return new WP_Error( 'image_mime_type_invalid', __( 'The provided mime type is not allowed.', 'performance-lab' ) );
+		}
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $targeted_mime ) ) ) {
+			return new WP_Error( 'image_mime_type_not_supported', __( 'The provided mime type is not supported.', 'performance-lab' ) );
+		}
+
+		$extension   = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$destination = trailingslashit( $original_directory ) . "{$filename}.{$extension[0]}";
+
+		$image->save( $destination, $targeted_mime );
+
+		$new_image = wp_get_image_editor( $destination );
+
+		if ( 'thumbnail' === $target ) {
+			$nocrop = true;
+		}
+
+		if ( isset( $sizes ) ) {
```

Comment:
An early return avoids an additional nesting in the code.

```suggestion
		if ( ! isset( $sizes ) ) {
		  continue;
		}
```